src/lib.rs

```diff
@@ -52,6 +52,7 @@ fn _pydantic_core(_py: Python, m: &PyModule) -> PyResult<()> {
     m.add_class::<PyUrl>()?;
     m.add_class::<PyMultiHostUrl>()?;
     m.add_class::<SchemaSerializer>()?;
+    m.add_class::<validators::ValidatorInfo>()?;
```

Comment:
Do we need this export?

Reply:
If we don't export it it would only be available with a `TYPE_CHECKING` guard. I think we should just export it.

Reply:
Not sure if this is likely to be important, but I think it also makes `isinstance` checks possible, figured that could be relevant.

Considering that, as far as I can tell, it's not possible to create instances of this type from python code, it seems okay to export it. Might make sense to give it a leading underscore if we want it to really be treated as private?

Reply:
Yep it won't be instantiable from Python. I'm not sure why anyone would want to do `isinstance` on it. The main thing I want by exporting it and making it a real class in Python is that users can do:

```python
from pydantic import ValidatorInfo

@validator("a")
def val_a(cls, v: str, info: ValidatorInfo) -> str: ...
```

Instead of:

```python
if TYPE_CHECKING:
  from pydantic import ValidatorInfo

@validator("a")
def val_a(cls, v: str, info: ValidatorInfo) -> str: ...
```

Reply:
I agree that it's nice to be able to do that and not need to add the if TYPE_CHECKING: if you aren't already using it.

Reply:
what we do for serializers is define the protocol in `core_schema.py`, see 

https://github.com/pydantic/pydantic-core/blob/5763f90e5eadcc750efb64a61415f0a85d0650a7/pydantic_core/core_schema.py#L53

That way you don't need the `if TYPE_CHECKING:`, and there's no import issue in `core_schema.py`.

I'm a bit torn, I think it would be better to do it the same way in both places, but I don't want to require this somewhat bikeshedy change.

@adriangb you can decide.

Reply:
I think this approach is more correct. Otherwise you are saying "this is the class" in one `pydantic_core.core_schema` but then `isinstance` and other introspection will say otherwise! I'd be okay with either making it a typing only class for Python (in which case you can't do isinstance) or a real class, but not a real class that is different from the actual class. I think we should change that approach elsewhere to match this.

Reply:
Not sure if I'm interpreting your comment correctly @adriangb (i.e., not sure what you mean when you say "this approach" — your current approach in the PR, or Samuel's suggestion), but for what it's worth, inheriting from `Protocol` does prevent you from doing an isinstance check:

```
TypeError: Instance and class checks can only be used with @runtime_checkable protocols
```

Reply:
True I guess inheriting from a protocol does bar those checks. I’m fine with that then, I didn’t pay full attention above.

Reply:
Moved to the suggested approach. Thanks for the patience!